gum/gummemory.c

```diff
@@ -263,54 +288,106 @@ gum_memory_scan (const GumMemoryRange * range,
 }
 
 GumMatchPattern *
-gum_match_pattern_new_from_string (const gchar * match_str)
+gum_match_pattern_new_from_string (const gchar * match_combined_str)
 {
-  GumMatchPattern * pattern;
+  GumMatchPattern * pattern = NULL;
+  gchar ** splitted;
+  const gchar * match_str, * mask_str;
+  gboolean has_mask = FALSE;
   GumMatchToken * token = NULL;
-  const gchar * ch;
+  const gchar * ch, * mh;
+
+  splitted = g_strsplit (match_combined_str, ":", 2);
+  match_str = splitted[0];
+  if (match_str == NULL)
+    goto parse_error;
+
+  mask_str = splitted[1];
+  has_mask = mask_str != NULL;
+  if (has_mask && strlen (mask_str) != strlen (match_str))
```

Comment:
This compares the inputs before any ignored spaces have been considered. I suppose we could fix this by moving the check later, but OTOH enforcing consistency is a good thing so we could also just leave this as it is.